src/View/AbstractDirectoryView.vala

```diff
@@ -558,8 +558,10 @@ namespace FM {
                         }
                     }
                 } else if (default_app != null) {
+                    var files_to_open = get_files_for_action ();
```

Comment:
```suggestion
                    /* Because this is in another thread we need to copy the selection to ensure it remains valid */
                    var files_to_open = selection.copy_full (GLib.Object.ref);
```
It is probably better like this so we are logically using the exact same list

Reply:
Do you mean this?
```
var files_to_open = selection.copy_deep ((GLib.CopyFunc)(GLib.Object.ref));
```

Reply:
I have followed your suggestion (using copy_deep)